csharp/src/Apache.Arrow/Ipc/IpcOptions.cs

```diff
@@ -25,6 +25,23 @@ public class IpcOptions
         /// </summary>
         public bool WriteLegacyIpcFormat { get; set; }
 
+        /// <summary>
+        /// The compression codec to use to compress data buffers.
+        /// If null (the default value), no compression is used.
+        /// </summary>
+        public CompressionCodecType? CompressionCodec { get; set; }
+
+        /// <summary>
+        /// The compression codec factory used to create compression codecs.
+        /// Must be provided if a CompressionCodec is specified.
+        /// </summary>
+        public ICompressionCodecFactory CompressionCodecFactory { get; set; }
```

Comment:
I'm not sure whether the factory should be included in the options or whether it makes more sense to be passed to the writer constructor. It's nice to have here to avoid complicating the constructor too much, and because it goes together with the codec option, but I don't know if it should really be considered an "option".

Reply:
Another approach might be to take an ICompressionCodec instead; that seems a bit more options-like than taking both the factory and the codec identity. This would probably require ICompressionCodec to expose a CompressionCodecType property, though.

I don't know that I feel strongly either way.

Reply:
Yeah that's true, although if you don't feel too strongly about it then I'm happy to leave it as is.